library/std/src/sys/unix/mod.rs

```diff
@@ -61,8 +61,11 @@ pub unsafe fn init(argc: isize, argv: *const *const u8) {
     // want!
     //
     // Hence, we set SIGPIPE to ignore when the program starts up in order
-    // to prevent this problem.
-    reset_sigpipe();
+    // to prevent this problem. Add #![no_ignore_sigpipe] to your
+    // bin crate to never start ignoring SIGPIPE.
+    if ignore_sigpipe {
+        reset_sigpipe();
+    }
```

Comment:
Is this code still codegened into binary with lto?

Reply:
I did a quick and probably naive test, but it seems to me as if this code is removed with `lto`.

With `#![ignore_sigpipe(no)]`:

```
% otool -tvV target/release/yes | grep signal
__ZN3std3sys4unix14stack_overflow3imp14signal_handler17h0422f13a4fc68dbdE:
000000010002529c        bl      0x1000336a4 ; symbol stub for: _pthread_cond_signal
```

With `#![ignore_sigpipe(yes)]`

```
% otool -tvV target/release/yes | grep signal
__ZN3std3sys4unix14stack_overflow3imp14signal_handler17h0422f13a4fc68dbdE:
0000000100025164        bl      0x10003364c ; symbol stub for: _pthread_cond_signal
00000001000306cc        bl      0x10003373c ; symbol stub for: _signal
```

Note that the symbol stub for `_signal` is not present with `#![ignore_sigpipe(no)]`.